File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.cpp
#include <string>
#include <vector>
#include <regex>

using namespace std;

vector<string> words({"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"});

int solution(string s) {
    for(int i = 0; i < 10; i++) {
        s = regex_replace(s, regex(words[i]), to_string(i));
    }
    
    return stoi(s);
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):    
    words = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]
    
    for number, word in enumerate(words):
        s = s.replace(word, str(number))
        
    answer = int(s)
    
    return answer